packages/11ty/content/_assets/javascript/application/canvas-panel.js

```diff
@@ -21,30 +21,40 @@ const annotationData = (input) => {
  * @param  {Array} annotationIds  The IIIF ids of the annotations to select
  * @param  {String} region      The canvas region
  */
-const goToCanvasState = function ({ annotationIds=[], figureId, region='' }) {
+const goToCanvasState = function ({ annotationIds=[], figureId, region }) {
   if (!figureId) return
-  const figure = document.querySelector(`#${figureId}, [data-lightbox-slide-id="${figureId}"]`)
-  if (!figure) return
-  const canvasPanel = figure.querySelector('canvas-panel')
-  if (region && canvasPanel.getAttribute('preset') !== 'zoom') {
-    console.warn(`Using the "annoref" shortcode to link to a region on a figure without zoom enabled is not supported. Please set the "preset" property to "zoom" on figure id "${figureId}"`)
-  }
-  const lightbox = figure.closest('q-lightbox')
-  const inputs = figure.querySelectorAll('.annotations-ui__input')
+  const figureSelector = `#${figureId}`
+  const slideSelector = `[data-lightbox-slide-id="${figureId}"]`
+  const figure = document.querySelector(figureSelector)
+  const figureSlide = document.querySelector(slideSelector)
+  if (!figure && !figureSlide) return
+  [figure, figureSlide].forEach((element) => {
+    if (!element) return
+    const canvasPanel = element.querySelector('canvas-panel')
+    if (region && canvasPanel.getAttribute('preset') !== 'zoom') {
+      console.warn(`Using the "annoref" shortcode to link to a region on a figure without zoom enabled is not supported. Please set the "preset" property to "zoom" on figure id "${figureId}"`)
+    }
+  })
+  const inputs = document.querySelectorAll(`#${figureId} .annotations-ui__input, [data-lightbox-slide-id="${figureId}"] .annotations-ui__input`)
   const annotations = [...inputs].map((input) => {
-    const id = input.getAttribute('value')
+    const id = input.getAttribute('data-annotation-id')
     input.checked = annotationIds.includes(id)
     return annotationData(input)
   })
 
-  if (lightbox) {
+  if (figureSlide) {
+    const lightbox = figureSlide.closest('q-lightbox')
     lightbox.currentId = figureId
   }
 
   /**
    * Update Canvas state
    */
-  update(canvasPanel, { annotations, region })
+  const canvasId = document
+    .querySelector(`${figureSelector}, ${slideSelector}`)
+    .querySelector('canvas-panel')
```

Comment:
An alternative way of selecting the `canvas-panel` element
```suggestion
    .querySelector(`[${figureSelector}, ${slideSelector}]:has('canvas-panel')`)
```

Reply:
Is there a use case for encapsulating this into a `getCanvasPanelId(element)` function that can handle this use and below in `handleSelect`?